test/built-ins/RegExp/prototype/dotAll/this-val-invalid-obj.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Invoked on an object without an [[OriginalFlags]] internal slot
+esid: pending
+info: >
+    get RegExp.prototype.dotAll
+
+    1. Let R be the this value.
+    2. If Type(R) is not Object, throw a TypeError exception.
+    3. If R does not have an [[OriginalFlags]] internal slot, throw a TypeError
+       exception.
+features: [regexp-dotall]
+---*/
+
+var dotAll = Object.getOwnPropertyDescriptor(RegExp.prototype, 'dotAll').get;
+
+assert.throws(TypeError, function() {
+  dotAll.call({});
+}, 'ordinary object');
+
+assert.throws(TypeError, function() {
+  dotAll.call([]);
+}, 'array exotic object');
+
+assert.throws(TypeError, function() {
+  dotAll.call(arguments);
```

Comment:
I just copied these tests from the other flag tests (e.g., see `test/built-ins/RegExp/prototype/global/this-val-invalid-obj.js`). What if this patch let them go in as is, and a separate patch refactored all of them?